src/VecSim/spaces/spaces.cpp

```diff
@@ -30,6 +31,19 @@ dist_func_t<float> GetDistFunc<vecsim_types::bfloat16, float>(VecSimMetric metri
     throw std::invalid_argument("Invalid metric");
 }
 
+template <>
+dist_func_t<float> GetDistFunc<vecsim_types::float16, float>(VecSimMetric metric, size_t dim,
+                                                             unsigned char *alignment) {
+    switch (metric) {
+    case VecSimMetric_Cosine:
+    case VecSimMetric_IP:
+        return IP_FP16_GetDistFunc(dim, nullptr, alignment);
```

Comment:
why does this function get a void pointer that is always null? can we just not pass it?

Reply:
It was part of the optimization bit map refactor
It was left mainly for testing purposes

Reply:
consider cleaning it now? I don't think it is needed for the test either (unless we suspect the hardware will change between calls)

Reply:
we pass a specific optimization in the tests to enforce lower optimization
see test_spaces.cpp